server/services/store/generators/main.go

```diff
@@ -0,0 +1,305 @@
+	out := bytes.NewBufferString("")
+	metadata, err := extractStoreMetadata()
+	if err != nil {
+		return nil, err
+	}
+	metadata.Name = name
+
+	myFuncs := template.FuncMap{
+		"joinResults": func(results []string) string {
+			return strings.Join(results, ", ")
+		},
+		"joinResultsForSignature": func(results []string) string {
+			if len(results) == 0 {
+				return ""
+			}
+			if len(results) == 1 {
+				return strings.Join(results, ", ")
+			}
+			return fmt.Sprintf("(%s)", strings.Join(results, ", "))
+		},
+		"genResultsVars": func(results []string, withNilError bool) string {
+			vars := []string{}
+			for i, typeName := range results {
+				switch {
+				case isError(typeName):
+					if withNilError {
+						vars = append(vars, "nil")
+					} else {
+						vars = append(vars, "err")
+					}
+				case i == 0:
+					vars = append(vars, "result")
+				default:
+					vars = append(vars, fmt.Sprintf("resultVar%d", i))
+				}
+			}
+			return strings.Join(vars, ", ")
+		},
+		"errorToBoolean": func(results []string) string {
+			for _, typeName := range results {
+				if isError(typeName) {
+					return "err == nil"
+				}
+			}
+			return "true"
+		},
+		"errorPresent": func(results []string) bool {
+			for _, typeName := range results {
+				if isError(typeName) {
+					return true
+				}
+			}
+			return false
+		},
+		"errorVar": func(results []string) string {
+			for _, typeName := range results {
+				if isError(typeName) {
+					return "err"
+				}
+			}
+			return ""
+		},
+		"joinParams": func(params []methodParam) string {
+			paramsNames := make([]string, 0, len(params))
+			for _, param := range params {
+				tParams := ""
+				if strings.HasPrefix(param.Type, "...") {
+					tParams = "..."
+				}
+				paramsNames = append(paramsNames, param.Name+tParams)
+			}
+			return strings.Join(paramsNames, ", ")
+		},
+		"joinParamsWithType": func(params []methodParam) string {
+			paramsWithType := []string{}
+			for _, param := range params {
+				switch param.Type {
+				case "Container":
+					paramsWithType = append(paramsWithType, fmt.Sprintf("%s store.%s", param.Name, param.Type))
+				default:
+					paramsWithType = append(paramsWithType, fmt.Sprintf("%s %s", param.Name, param.Type))
+				}
+			}
+			return strings.Join(paramsWithType, ", ")
+		},
+		"joinParamsWithTypeOutsideStore": func(params []methodParam) string {
+			paramsWithType := []string{}
+			for _, param := range params {
+				switch param.Type {
+				case "Container":
+					paramsWithType = append(paramsWithType, fmt.Sprintf("%s store.%s", param.Name, param.Type))
+				default:
+					paramsWithType = append(paramsWithType, fmt.Sprintf("%s %s", param.Name, param.Type))
+				}
+			}
+			return strings.Join(paramsWithType, ", ")
+		},
+		"renameStoreMethod": func(methodName string) string {
+			return strings.ToLower(methodName[0:1]) + methodName[1:]
+		},
+		"genErrorResultsVars": func(results []string, errName string) string {
```

Comment:
I have some store methods for notifications that use the Channels style methods where a create/insert will return a pointer to the struct that was inserted. e.g. 
```
func (s *SQLStore) UpsertNotificationHint(hint *model.NotificationHint) (*model.NotificationHint, error) 
```
Is it correct that pattern won't work for this generator?

Reply:
Miguel confirmed any return types can be used.